lib/graphql/batch/execution_strategy.rb

```diff
@@ -49,7 +49,13 @@ def each_promise_in_entry(obj, key, value, &block)
     class FieldResolution < GraphQL::Query::SerialExecution::FieldResolution
       def get_finished_value(raw_value)
         if raw_value.is_a?(::Promise)
-          raw_value.then { |result| super(result) }
+          on_reject = lambda do |error|
+            raise error unless error.is_a?(GraphQL::ExecutionError)
+            error.ast_node = ast_node
+            query.context.errors << error
+            nil
+          end
+          raw_value.then { |result| super(result) }.then(nil, on_reject)
```

Comment:
I guess we don't really have to worry about GraphQL::ExecutionError being raised from get_finished_value, so I can probably use the single `.then`

I think having a `.rescue` or `.catch` in promise.rb would make it more readable to specify an on_reject callback.